spec/lib/metadata.spec.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-invalid-this */
+"use strict";
+
+describe("lib/metadata", () => {
+    var authToken, bluebird, error, instance, logger, Metadata, requestOptionsMock, requestPromiseMock, responseHandlerMock, uri;
+
+    bluebird = require("bluebird");
+    error = require("../../lib/error")();
+    logger = require("../../lib/logger")();
+    requestOptionsMock = require("../mock/request-options-mock")();
+    requestPromiseMock = require("../mock/request-promise-mock")();
+    responseHandlerMock = require("../mock/response-handler-mock")();
+    Metadata = require("../../lib/metadata")(bluebird, error, logger, requestOptionsMock, requestPromiseMock, responseHandlerMock);
+    beforeEach(() => {
+        instance = new Metadata(uri, authToken);
+        spyOn(bluebird, "reject").andCallThrough();
+    });
+    [
+        {
+            fn: "getAll",
+            verb: "get"
+        }, {
+            fn: "getProperty",
```

Comment:
To illustrate my point, i've updated `getProperty` so that it would be very broken but still pass all the tests.

```
         getProperty(name) {
-            var options;
-
-            logger.debug(`In ${this.constructor.name}.getProperty()`);
-            logger.debug(`Name param: ${name}`);
-
             if (!name) {
-                return bluebird.reject(new Error("Must provide the name for the property to get.", error.INCORRECT_PARAMETERS));
+                return bluebird.reject("lol error");
             }
 
-            options = requestOptions.createOptions(this.authToken);
-            options.json = true;
-            options.url = `${this.uri}/${name}`;
-            logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
-
-            return requestPromise.get(options).then((response) => {
-                logger.debug(`In ${this.constructor.name}.getProperty() -> request.get().then()`);
-                logger.debug(`Response: ${JSON.stringify(response, null, 4)}`);
-                logger.debug(`Status code: ${response.statusCode}`);
-                logger.debug(`Response body: ${JSON.stringify(response.body, null, 4)}`);
-
-                return response.body;
-            }, (reason) => {
-                logger.debug(`In ${this.constructor.name}.getProperty() -> request.get().catch()`);
-                logger.debug(`Reason: ${JSON.stringify(reason, null, 4)}`);
+            requestOptions.createOptions("LOL AUTH TOKEN");
 
+            return requestPromise.get().then(() => {
+                return null;
+            }, () => {
                 // Pass the reason response to the error handler.
-                responseHandler.handleErrorResponse(reason);
+                responseHandler.handleErrorResponse();
             });
         }

```